internal/services/datafactory/data_factory.go

```diff
@@ -97,6 +97,31 @@ func flattenDataFactoryParameters(input map[string]*datafactory.ParameterSpecifi
 	return output
 }
 
+func expandDataFactoryParametersFourPointOh(input []interface{}) map[string]*datafactory.ParameterSpecification {
+	parameters := make(map[string]*datafactory.ParameterSpecification)
+	for _, v := range input {
+		val := v.(map[string]interface{})
+		parameters[val["name"].(string)] = &datafactory.ParameterSpecification{
+			Type:         datafactory.ParameterType(val["type"].(string)),
+			DefaultValue: val["default_value"],
```

Comment:
Is it necessary to convert the `default value` to the specified `type`?

Reply:
No, we can pass the string value, and the API can do the conversion, if it can't be converted, it will be set as String (and the `type` stays at the value whatever we pass to it)